lib/web_ui/lib/src/engine/window.dart

```diff
@@ -707,34 +707,34 @@ class EngineWindow extends ui.Window {
   }
 
   @override
-  void render(ui.Scene/*!*/ scene) {
+  void render(ui.Scene scene) {
     if (experimentalUseSkia) {
-      final LayerScene layerScene = scene;
-      rasterizer.draw(layerScene.layerTree);
+      final LayerScene layerScene = scene as LayerScene;
+      rasterizer!.draw(layerScene.layerTree);
     } else {
-      final SurfaceScene surfaceScene = scene;
+      final SurfaceScene surfaceScene = scene as SurfaceScene;
       domRenderer.renderScene(surfaceScene.webOnlyRootElement);
     }
   }
 
   @visibleForTesting
-  Rasterizer rasterizer = experimentalUseSkia ? Rasterizer(Surface()) : null;
+  late Rasterizer? rasterizer = experimentalUseSkia ? Rasterizer(Surface(HtmlViewEmbedder())) : null;
 }
 
-bool _handleWebTestEnd2EndMessage(MethodCodec codec, ByteData data) {
+bool _handleWebTestEnd2EndMessage(MethodCodec codec, ByteData? data) {
   final MethodCall decoded = codec.decodeMethodCall(data);
   double ratio = double.parse(decoded.arguments);
   switch (decoded.method) {
     case 'setDevicePixelRatio':
       window.debugOverrideDevicePixelRatio(ratio);
-      window.onMetricsChanged();
+      window.onMetricsChanged!();
```

Comment:
non-null

Reply:
This is allowed by the framework to be null.